docs/customization/controller.rst

```diff
@@ -0,0 +1,167 @@
+getting a list of recommended product from your external api.
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Controller;
+
+    use FOS\RestBundle\View\View;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
+    use Sylius\Bundle\CoreBundle\Controller\ProductController as BaseProductController;
+    use Sylius\Component\Resource\ResourceActions;
+
+    class ProductController extends BaseProductController
+    {
+        /**
+         * @param Request $request
+         *
+         * @return Response
+         */
+        public function showAction(Request $request)
+        {
+            $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+            $this->isGrantedOr403($configuration, ResourceActions::SHOW);
+            $product = $this->findOr404($configuration);
+
+            $recommendationServiceApi = $this->get('app.recommendation_service_api');
+
+            $recommendedProducts = $recommendationServiceApi->getRecommendedProducts($product);
+
+            $this->eventDispatcher->dispatch(ResourceActions::SHOW, $configuration, $product);
+
+            $view = View::create($product);
+
+            if ($configuration->isHtmlRequest()) {
+                $view
+                    ->setTemplate($configuration->getTemplate(ResourceActions::SHOW . '.html'))
+                    ->setTemplateVar($this->metadata->getName())
+                    ->setData([
+                        'configuration' => $configuration,
+                        'metadata' => $this->metadata,
+                        'resource' => $product,
+                        'recommendedProducts' => $recommendedProducts,
+                        $this->metadata->getName() => $product,
+                    ])
+                ;
+            }
+
+            return $this->viewHandler->handle($configuration, $view);
+        }
+    }
+
+2. In order to use your controller and its actions you need to configure it in the ``app/config/config.yml``.
+
+.. code-block:: yaml
+
+    sylius_product:
+        resources:
+            product:
+                classes:
+                    controller: AppBundle\Controller\ProductController
+
+How to customize a Standard Controller?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would like to send some kind of emails (which are not resources) after something has been purchased in your shop - to do this you should modify an ``afterPurchaseAction`` on the ``OrderController``.
+
+1. Create a new Controller class under the ``AppBundle/Controller/Frontend`` namespace.
+
+Run ``$ php app/console debug:container sylius.controller.frontend.order``.
+
+Your class needs to be extending this base class.
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Controller\Frontend;
+
+    use Sylius\Bundle\WebBundle\Controller\Frontend\Account\OrderController as BaseOrderController;
```

Comment:
Is it responsible to use in example class from *WebBundle* if it'll probably be 💀 in the nearest future?